public/app/core/components/AppChrome/TopBar/TopBarExtensionPoint.tsx

```diff
@@ -0,0 +1,35 @@
+import { useContext } from 'react';
+
+import { renderLimitedComponents } from '@grafana/runtime';
+import { AddedComponentsRegistryContext } from 'app/features/plugins/extensions/ExtensionRegistriesContext';
+import { usePluginComponents } from 'app/features/plugins/extensions/usePluginComponents';
+
+/**
+ * Extension point for plugins to add components to the top bar.
+ * Currently restricted to grafana-setupguide-app plugin.
+ */
+export function TopBarExtensionPoint() {
+  // Check if plugin extension context is available (not available in tests)
+  const context = useContext(AddedComponentsRegistryContext);
+
+  if (!context) {
+    // Silently return null in test environments or when plugin context is not available
+    return null;
+  }
+
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const { components } = usePluginComponents({
+    extensionPointId: 'grafana/singletopbar/action',
```

Comment:
Maybe move this one to the enum in `@grafana/data`.

https://github.com/grafana/grafana/blob/07d975134d27f1c6bf44e326511bbef1bd5af26b/packages/grafana-data/src/types/pluginExtensions.ts#L199